Assets/__Scripts/MapEditor/Grid/Collections/BeatmapObjectContainerCollection.cs

```diff
@@ -358,9 +358,14 @@ public abstract void SpawnObject(BaseObject obj, out List<BaseObject> conflictin
 
     public static void RefreshFutureObjectsPosition(float jsonTime)
     {
-        foreach (var objectType in System.Enum.GetValues(typeof(Beatmap.Enums.ObjectType)))
+        // we have to refresh bpm events FIRST, and only then can we refresh other objects
+        // this is a janky way of accomplishing that
+        var objectTypes = new List<ObjectType> { ObjectType.BpmChange };
+        objectTypes.AddRange(((IEnumerable<ObjectType>)Enum.GetValues(typeof(ObjectType))).Where(x => x != ObjectType.BpmChange));
```

Comment:
At this point, considering we are likely not to add any future `ObjectType`s for quite a while, if at all... would it make more sense just to hardcode this list?

Reply:
yeah I guess I'll just do that